HackIllinois/DataSources/HIEventDataSource.swift

```diff
@@ -27,46 +27,39 @@ final class HIEventDataSource {
         }
         isRefreshing = true
 
-        HIEventService.getAllLocations()
-        .onCompletion { result in
-            if case let .success(containedLocations, _) = result {
-                print("GET::LOCATIONS::SUCCESS")
-                HIEventService.getAllEvents()
-                .onCompletion { result in
-                    if case let .success(containedEvents, _) = result {
-                        print("GET::EVENTS::SUCCESS")
-                        HIEventService.getAllFavorites()
-                        .onCompletion { result in
-                            if case let .success(containedFavorites, _) = result {
-                                print("GET::FAVORITES::SUCCESS")
-
-                                var updatedEvents = [HIAPIEvent]()
-                                containedEvents.data.forEach { event in
-                                    var event = event
-                                    event.favorite = containedFavorites.data.map { $0.eventId }.contains(event.id)
-                                    updatedEvents.append(event)
-                                }
-                                setEventNotifications(containedLocations: containedLocations, updatedEvents: updatedEvents)
-                            }
-                            completion?()
-                            isRefreshing = false
-                        }
-                        .authorize(with: HIApplicationStateController.shared.user)
-                        .launch()
-                    } else {
-                        completion?()
-                        isRefreshing = false
+        HIEventService.getAllEventsRaw(completed: {events in
```

Comment:
Do you not need to fetch locations or favorites anymore

Reply:
dont need to fetch locations anymore, working on favorites